actionview/lib/action_view/helpers/tag_helper.rb

```diff
@@ -164,6 +330,10 @@ def respond_to_missing?(*args)
           end
 
           def method_missing(called, *args, **options, &block)
+            method_name = TagBuilder.define_element(called.to_s.dasherize, method_name: called)
+
+            public_send(method_name, *args, **options, &block)
+          rescue SyntaxError
```

Comment:
Unfortunately, this is necessary to pass tests that attempt to render elements with [COMMON_DANGEROUS_CHARS = "&<>\"' %*+,/;=^|"](https://github.com/rails/rails/blob/6ef9007cf5da7392280b15becd6948dd6de8eb06/actionview/test/template/tag_helper_test.rb#L10), since the characters aren't valid Ruby.

If https://github.com/rails/rails/pull/49175 gets merged, failing fast will remove any name safety considerations from this method.

Reply:
I've rebased off the latest, so this is no longer necessary.